insights/collect.py

```diff
@@ -361,23 +361,40 @@ def create_archive(path, remove_path=True):
     return archive_path
 
 
-def collect(manifest=default_manifest, tmp_path=None, compress=False, rm_conf=None, client_timeout=None):
+def generate_relative_path():
+    """
+    Creates the relative path to store the component output.
+    the original path is removed.
+    """
+    try:
+        hostname = call("hostname -f", env=SAFE_ENV).strip()
+    except CalledProcessError:
+        # problem calling hostname -f
+        hostname = call("hostname", env=SAFE_ENV).strip()
+    suffix = datetime.utcnow().strftime("%Y%m%d%H%M%S")
+    return "insights-%s-%s" % (hostname, suffix)
+
+
+def collect(manifest=default_manifest, tmp_path=None, compress=False,
```

Comment:
@bfahr - the `Traceback` looks a bit weird, I guess you might be on a different local branch? Since the file `insights/specs/datasources/forever.py` does not exist in this PR. I tried it on my RHEL 8/9 test env but didn't meet this issue. Or, may I know how to do you `"run collect from insights shell"`?

And, Thanks for reviewing!

Reply:
@xiangce you are correct, that is a test datasource that I had not removed before testing your branch.  I ran again on RHEL8 and it works correctly.